### PR TITLE
Undeploy command

### PR DESCRIPTION
**Pull request by lindnerby**
**Description**
For the new Kyma 2 command `deploy` we need a corresponding `undeploy` command. The first iteration of this command just removes namespaces and CRDs.

Changes proposed in this pull request:

- Introduce new command `undeploy`
- Removes finalizers from CRs that are managed by reconciler and istio
- Removes namespaces `kyma-system`, `istio-system`, `kyma-integration`
- Removes Kyma CRDs

**Related issue(s)**

* https://github.com/kyma-project/kyma/issues/12140

### CODE REVIEW

**Comment by jeremyharisch** [comment]
When using `kyma undeploy` the namespace kyma-system gets stuck in the state `Terminating`.

```bash
kubectl get ns
[...]
kyma-system       Terminating   75m
[...]
```

File: cmd/kyma/undeploy/cmd.go [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+	if err := cmd.removeCustomResourcesFinalizers(); err != nil {
+		step.Failure()
+		return err
+	}
+
+	step.Successf("Finalizers successfully removed")
+	return nil
+}
+
+func (cmd *command) removeServerlessCredentialsFinalizers() error {
```
[the remark this answers is not in the document]
Comment (lindnerby) [reply cmd/kyma/undeploy/cmd.go:113]: That's the same name. What did you have in mind?

File: cmd/kyma/undeploy/cmd.go [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+		RunE:  func(_ *cobra.Command, _ []string) error { return cmd.Run() },
+	}
+
+	cobraCmd.Flags().BoolVarP(&o.KeepCRDs, "keep-crds", "", false, "Set --keep-crds=true to keep CRDs on clean-up")
+	return cobraCmd
+}
+
+//Run runs the command
+func (cmd *command) Run() error {
+	var err error
```
[the remark this answers is not in the document]
Comment (lindnerby) [reply cmd/kyma/undeploy/cmd.go:63]: Afaik you can not create new steps within a new step. It's also consistent to other commands, so I would now keep it as is.

**Comment by lindnerby** [comment]
> When using `kyma undeploy` the namespace kyma-system gets stuck in the state `Terminating`.
> 
> ```shell
> kubectl get ns
> [...]
> kyma-system       Terminating   75m
> [...]
> ```

So when removing finalizers two times, it's now terminating after one minute. There seems to be some exp backoff for a finalizer of pvc-protection... not sure if that's acceptable.

File: cmd/kyma/undeploy/cmd.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyharisch) [reply cmd/kyma/undeploy/cmd.go:116]: ups XD Copy/paste error `removeServerlessCredentialFinalizers`, without the `s`

File: cmd/kyma/undeploy/cmd.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremyharisch) [reply cmd/kyma/undeploy/cmd.go:63]: When you look [here](https://github.com/kyma-project/cli/blob/main/cmd/kyma/alpha/deploy/cmd.go#L170) we created the Start Kyma deploying step, and this ends when Kyma deployment succeeded or failed, and between start and success/failure we have a [new step](https://github.com/kyma-project/cli/blob/main/cmd/kyma/alpha/deploy/cmd.go#L197) for each component.

[the remark this answers is not in the document]
Comment (jeremyharisch) [reply cmd/kyma/undeploy/cmd.go:63]: We leave it as it is for now, as discussed in private

**Review by NHingerl — APPROVED** [review]
We must remember to update docs/04-operation-guides/operations/07-uninstall-kyma as well

**Review by jeremyharisch — APPROVED** [review]

**Pull request merged**